LuatixHQ/opencti
# Prevent input name collision with db name for refs (#6056)

Several refs attributes have the same input name as their db name, resulting in bugs during stream event publishing.

If the attribute is multiple, the event published in stream contains `null` in patch.
If it's a single value, the stream event won't be sent and an error will be thrown (`store update event error`)

### Proposed changes

* on schema registration, check the names collisions and throw an error 
* fix all collisions found by prefixing the name
* rename API keys that require changes (--> only `MalwareAnalysisAddInput.sample` to rename `MalwareAnalysisAddInput.analysisSample`

### Related issues

* #6056 

### Checklist


### Further comments

e2e tests would be added in #6073 to cover network traffic creation + ref, I won't add it here (or improve on it here once merge and rebase)

To test this PR, test the following:
* open the raw stream at http://localhost:3000/stream
* Create an observable > network traffic, with a random dst port
* Open it, go to knowledge, and add a nested object
* check single ref
  * select an ipv4 address, and create a "src" relationship (not dst, you will end up in #6070)
  * BEFORE --> store update event, NOW -> no error
  * check in stream the update event is correct
* check multiple ref
  * create 2 other network traffic using just some random DST port values
  * select the inital network traffic, and create 2 "encapsulates" relationship with the 2 other network traffic
  * BEFORE --> no store update event but NULL in the stream events, NOW -> no error, correct values in the stream events
